cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/increment/hbase11/IncrementSummingScanner.java

```diff
@@ -0,0 +1,368 @@
+    this.region = region;
+    this.batchSize = batchSize;
+    this.baseScanner = new WrappedScanner(internalScanner);
+    if (internalScanner instanceof RegionScanner) {
+      this.baseRegionScanner = (RegionScanner) internalScanner;
+    }
+    this.scanType = scanType;
+    this.compactionUpperBound = compationUpperBound;
+    this.oldestTsByTTL = oldestTsByTTL;
+  }
+
+  @Override
+  public HRegionInfo getRegionInfo() {
+    return region.getRegionInfo();
+  }
+
+  @Override
+  public boolean isFilterDone() throws IOException {
+    if (baseRegionScanner != null) {
+      return baseRegionScanner.isFilterDone();
+    }
+    throw new IllegalStateException(
+      "RegionScanner.isFilterDone() called when the wrapped scanner is not a RegionScanner");
+  }
+
+  @Override
+  public boolean reseek(byte[] bytes) throws IOException {
+    if (baseRegionScanner != null) {
+      return baseRegionScanner.reseek(bytes);
+    }
+    throw new IllegalStateException(
+      "RegionScanner.reseek() called when the wrapped scanner is not a RegionScanner");
+  }
+
+  @Override
+  public long getMaxResultSize() {
+    if (baseRegionScanner != null) {
+      return baseRegionScanner.getMaxResultSize();
+    }
+    throw new IllegalStateException(
+      "RegionScanner.isFilterDone() called when the wrapped scanner is not a RegionScanner");
+  }
+
+
+  @Override
+  public long getMvccReadPoint() {
+    if (baseRegionScanner != null) {
+      return baseRegionScanner.getMvccReadPoint();
+    }
+    throw new IllegalStateException(
+      "RegionScanner.isFilterDone() called when the wrapped scanner is not a RegionScanner");
+  }
+
+  /**
+   * @return The limit on the number of cells to retrieve on each call to next(). See
+   * {@link Scan#setBatch(int)}
+   */
+  @Override
+  public int getBatch() {
+    return batchSize;
+  }
+
+  @Override
+  public boolean nextRaw(List<Cell> cells) throws IOException {
+    return nextRaw(cells, ScannerContext.newBuilder().setBatchLimit(batchSize).build());
+  }
+
+  @Override
+  public boolean nextRaw(List<Cell> cells, ScannerContext scannerContext) throws IOException {
+    return nextInternal(cells, scannerContext);
+  }
+
+  @Override
+  public boolean next(List<Cell> cells) throws IOException {
+    return next(cells, ScannerContext.newBuilder().setBatchLimit(batchSize).build());
+  }
+
+  @Override
+  public boolean next(List<Cell> cells, ScannerContext scannerContext) throws IOException {
+    return nextInternal(cells, scannerContext);
+  }
+
+  private boolean nextInternal(List<Cell> cells, ScannerContext scannerContext) throws IOException {
+    if (LOG.isTraceEnabled()) {
+      LOG.trace("nextInternal called with limit=" + scannerContext);
+    }
+    Cell previousIncrement = null;
+    long runningSum = 0;
+    int addedCnt = 0;
+    baseScanner.startNext();
+    Cell cell;
+    while ((cell = baseScanner.peekNextCell(getBatch())) != null && (getBatch() <= 0 || addedCnt < getBatch())) {
```

Comment:
We need to use limit from `scannerContext` everwhere rather than `getBatch` when `scannerContext` is available. 

Actually, I think the newly introduced method `getBatch` is not required at all. Any reason for adding it?